### ISSUE TITLE
Can not manage dates in a newly added month

### ISSUE THREAD

**Issue by ArtyomResh** [airbnb/react-dates#700]
Hi, everyone! 

That happens when you change a number of months via `numberOfMonths` props right in storybook, no matter how (via wrapper element with state or knobs).

![calendar_bug](https://user-images.githubusercontent.com/20782623/30050311-b87fea54-9226-11e7-9609-24afca789909.gif)

You can't hover or click on those dates at all. Seems it only happens when you change a number of months in real time.
I managed to reproduce this error in your examples. Version: `12.5.1`. The code I'm using:
```
/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import momentPropTypes from 'react-moment-proptypes';
import { forbidExtraProps } from 'airbnb-prop-types';
import moment from 'moment';
import omit from 'lodash/omit';

import DayPickerRangeController from '../src/components/DayPickerRangeController';

import ScrollableOrientationShape from '../src/shapes/ScrollableOrientationShape';

import { START_DATE, END_DATE, HORIZONTAL_ORIENTATION } from '../constants';
import isInclusivelyAfterDay from '../src/utils/isInclusivelyAfterDay';

const propTypes = forbidExtraProps({
  // example props for the demo
  autoFocusEndDate: PropTypes.bool,
  initialStartDate: momentPropTypes.momentObj,
  initialEndDate: momentPropTypes.momentObj,

  keepOpenOnDateSelect: PropTypes.bool,
  minimumNights: PropTypes.number,
  isOutsideRange: PropTypes.func,
  isDayBlocked: PropTypes.func,
  isDayHighlighted: PropTypes.func,

  // DayPicker props
  enableOutsideDays: PropTypes.bool,
  numberOfMonths: PropTypes.number,
  orientation: ScrollableOrientationShape,
  withPortal: PropTypes.bool,
  initialVisibleMonth: PropTypes.func,
  renderCalendarInfo: PropTypes.func,

  navPrev: PropTypes.node,
  navNext: PropTypes.node,

  onPrevMonthClick: PropTypes.func,
  onNextMonthClick: PropTypes.func,
  onOutsideClick: PropTypes.func,
  renderDay: PropTypes.func,

  // i18n
  monthFormat: PropTypes.string,

  isRTL: PropTypes.bool,
});

const defaultProps = {
  // example props for the demo
  autoFocusEndDate: false,
  initialStartDate: null,
  initialEndDate: null,

  // day presentation and interaction related props
  renderDay: null,
  minimumNights: 1,
  isDayBlocked: () => false,
  isOutsideRange: day => !isInclusivelyAfterDay(day, moment()),
  isDayHighlighted: () => false,
  enableOutsideDays: false,

  // calendar presentation and interaction related props
  orientation: HORIZONTAL_ORIENTATION,
  withPortal: false,
  initialVisibleMonth: null,
  numberOfMonths: 2,
  onOutsideClick() {},
  keepOpenOnDateSelect: false,
  renderCalendarInfo: null,
  isRTL: false,

  // navigation related props
  navPrev: null,
  navNext: null,
  onPrevMonthClick() {},
  onNextMonthClick() {},

  // internationalization
  monthFormat: 'MMMM YYYY',
};

class DayPickerRangeControllerWrapper extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      focusedInput: props.autoFocusEndDate ? END_DATE : START_DATE,
      startDate: props.initialStartDate,
      endDate: props.initialEndDate,
      numberOfMonths: 2,
    };

    this.onDatesChange = this.onDatesChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  onDatesChange({ startDate, endDate }) {
    this.setState({ startDate, endDate });
  }

  onFocusChange(focusedInput) {
    this.setState({
      // Force the focusedInput to always be truthy so that dates are always selectable
      focusedInput: !focusedInput ? START_DATE : focusedInput,
    });
  }

  handleChange(event) {
    this.setState({
      numberOfMonths: Number(event.target.value),
    });
  }

  render() {
    const { showInputs } = this.props;
    const { focusedInput, startDate, endDate } = this.state;

    const props = omit(this.props, [
      'autoFocus',
      'autoFocusEndDate',
      'initialStartDate',
      'initialEndDate',
    ]);

    const startDateString = startDate && startDate.format('YYYY-MM-DD');
    const endDateString = endDate && endDate.format('YYYY-MM-DD');

    return (
      <div>
        {showInputs &&
          <div style={{ marginBottom: 16 }}>
            <input type="text" name="start date" value={startDateString} readOnly />
            <input type="text" name="end date" value={endDateString} readOnly />
          </div>
        }
        <input type="number" value={this.state.numberOfMonths} onChange={this.handleChange} />

        <DayPickerRangeController
          {...props}
          numberOfMonths={this.state.numberOfMonths}
          onDatesChange={this.onDatesChange}
          onFocusChange={this.onFocusChange}
          focusedInput={focusedInput}
          startDate={startDate}
          endDate={endDate}
        />
      </div>
    );
  }
}

DayPickerRangeControllerWrapper.propTypes = propTypes;
DayPickerRangeControllerWrapper.defaultProps = defaultProps;

export default DayPickerRangeControllerWrapper;

```
error text: `Uncaught TypeError: Cannot read property '2017-12-01' of undefined`

I found a closed issue with same error, but different problem, that looks like has been solved, but I will share it with you anyway in case it can help you: https://github.com/airbnb/react-dates/issues/569

**Comment by majapw** [comment]
Thank you for including your code! I believe the issue is that the `componentWillReceiveProps` method of the `DayPickerRangeController` does not expand the state based on a change in `numberOfMonths`. 

If you look at the code here - https://github.com/airbnb/react-dates/blob/master/src/components/DayPickerRangeController.jsx#L217-L234, it looks like it would require a change in focus (from hidden to visible) to rerun the state generation code.

I think this makes sense for `initialVisibleMonth` but not for the other two props. I will put up a PR with the fix.

**Comment by majapw** [comment]
Note that in your gif the height also doesn't expand for this case so we'll need to cover that case as well.

**Comment by majapw** [comment]
This should be addressed in v12.6.0

**Comment by ArtyomResh** [comment]
Thank you! :D
